### PR TITLE
Implement the Any2StringAdd rewrite

### PR DESCRIPTION
**Pull request by dwijnand**
Part of https://github.com/scala/scala-dev/issues/327

### CODE REVIEW

File: rewrites/src/main/scala/fix/scala213/Any2StringAdd.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+final class Any2StringAdd extends SemanticRule("fix.scala213.Any2StringAdd") {
+  import Any2StringAdd._
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    doc.tree.collect {
+      case Term.ApplyInfix(lhs, Term.Name("+"), _, List(_)) if
+          lhs.synthetic.collect {
+            case ApplyTree(TypeApplyTree(IdTree(info), _), _) => info.symbol
+          }.contains(any2stringadd)
```
Comment (lrytz) [inline rewrites/src/main/scala/fix/scala213/Any2StringAdd.scala]: cool that it works with finding the synthetic `any2stringadd`

File: rewrites/src/main/scala/fix/scala213/Any2StringAdd.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    doc.tree.collect {
+      case plusString(Term.ApplyInfix(lhs, _, _, _)) => addToString(lhs)
+    }.asPatch
+  }
+
+  private def addToString(term: Term) = term match {
+    case _: Term.Name | _: Term.Select | _: Term.Block => Patch.addRight(term, ".toString")
+    case _ => Patch.addLeft(term, "(") + Patch.addRight(term, ").toString")
```
Comment (NthPortal) [inline rewrites/src/main/scala/fix/scala213/Any2StringAdd.scala]: my only concern here is that it unnecessarily boxes primitives.

Comment (dwijnand) [reply rewrites/src/main/scala/fix/scala213/Any2StringAdd.scala]: Ah, that must be why the deprecation message for these is to use string interpolation.

Should we do `"" + x` for these?  (And `"" + (...)` for non-Name/Select/Block.) . Or should we try to rewrite it to use string interpolation?

Comment (NthPortal) [reply rewrites/src/main/scala/fix/scala213/Any2StringAdd.scala]: I don't know. IIRC, string interpolation is special-cased by the compiler in 2.13, so it's probably at least as efficient as other options, if not more so?

Comment (dwijnand) [reply rewrites/src/main/scala/fix/scala213/Any2StringAdd.scala]: Yeah it's the right choice.  But for mechanical rewrites you could get some weird rewrites, so maybe this is best?

Comment (xuwei-k) [reply rewrites/src/main/scala/fix/scala213/Any2StringAdd.scala]: > that must be why the deprecation message for these is to use string interpolation.

> IIRC, string interpolation is special-cased by the compiler in 2.13


Yes. see https://github.com/scala/scala/pull/7201 https://github.com/scala/bug/issues/11025

**Pull request merged**